`test_jemalloc_percpu_arena/test.py::test_jemalloc_percpu_arena` is flaky

`Important: This issue was automatically created and is used by CI. Do not modify the issue title. Do not remove issue labels.`

Test name: test_jemalloc_percpu_arena/test.py::test_jemalloc_percpu_arena
Failure reason: DID NOT RAISE <class 'subprocess.CalledProcessError'>
CI report: [Integration tests (arm_binary, distributed plan, 4/4)](https://s3.amazonaws.com/clickhouse-test-reports/json.html?PR=92004&sha=bcbd687494d7bc799708dfdc5f1275f366bc096b&name_0=PR&name_1=Integration%20tests%20%28arm_binary%2C%20distributed%20plan%2C%204%2F4%29)

[CI DB](https://play.clickhouse.com/play?user=play#U0VMRUNUIGNoZWNrX3N0YXJ0X3RpbWUsIGNoZWNrX25hbWUsIHRlc3RfbmFtZSwgdGVzdF9zdGF0dXMsIGNoZWNrX3N0YXR1cywgcmVwb3J0X3VybApGUk9NIGNoZWNrcwpXSEVSRSAxCiAgICBBTkQgdGVzdF9zdGF0dXMgIT0gJ1NLSVBQRUQnCiAgICBBTkQgdGVzdF9zdGF0dXMgIT0gJ09LJwogICAgQU5EIGNoZWNrX3N0YXR1cyAhPSAnc3VjY2VzcycKICAgIEFORCB0ZXN0X25hbWUgaWxpa2UgJyV0ZXN0X2plbWFsbG9jX3BlcmNwdV9hcmVuYSUnCiAgICBBTkQgY2hlY2tfc3RhcnRfdGltZSA+IHRvZGF5KCkgLSBJTlRFUlZBTCAxMCBEQVkKT1JERVIgQlkgY2hlY2tfc3RhcnRfdGltZSBkZXNjLCBjaGVja19uYW1lLCB0ZXN0X25hbWU=)

```
    def test_jemalloc_percpu_arena():
        skip_if_jemalloc_disabled()
    
        assert multiprocessing.cpu_count() > CPU_ID
    
        online_cpus = int(run_with_cpu_limit("getconf _NPROCESSORS_ONLN"))
        assert online_cpus == 1, online_cpus
    
        all_cpus = int(run_with_cpu_limit("getconf _NPROCESSORS_CONF"))
        assert all_cpus == multiprocessing.cpu_count(), all_cpus
    
        # implicitly disable percpu arena
        result = run_with_cpu_limit(
            'clickhouse local -q "select 1"',
            # NOTE: explicitly disable, since it is enabled by default in debug build
            # (and even though debug builds are not in CI let's state this).
            "--env",
            "MALLOC_CONF=abort_conf:false",
        )
        assert int(result) == int(1), result
    
        # should fail because of abort_conf:true
>       with pytest.raises(subprocess.CalledProcessError):
E       Failed: DID NOT RAISE <class 'subprocess.CalledProcessError'>
```

--- a/tests/integration/test_jemalloc_percpu_arena/test.py
+++ b/tests/integration/test_jemalloc_percpu_arena/test.py
@@ -87,10 +87,13 @@ def test_jemalloc_percpu_arena():
     )
     assert int(result) == int(1), result
 
-    # should fail because of abort_conf:true
-    # This needs to pin the CPU > 0 manually because docked doesn't do it and
-    # /sys/devices/system/cpu/online is not enough.
+    # jemalloc compares sysconf(_SC_NPROCESSORS_ONLN) and sysconf(_SC_NPROCESSORS_CONF) to determine the CPU topology info
+    # Apparently sometimes _SC_NPROCESSORS_ONLN SOMETIMES ignores the override of /sys/devices/system/cpu/online and it should be equal to
+    # _SC_NPROCESSORS_CONF
+    # Latter it also checks sched_getaffinity() + CPU_COUNT(&set), so, we rely also on --cpuset-cpus (taskset) in order to trigger the
+    # second condition
     with pytest.raises(subprocess.CalledProcessError):
+        # should fail because of abort_conf:true but:
         run_with_cpu_limit(
             'clickhouse local -q "select 1"',
             "--cpuset-cpus", f"{CPU_ID}",

--- a/tests/integration/test_jemalloc_percpu_arena/test.py
+++ b/tests/integration/test_jemalloc_percpu_arena/test.py
@@ -46,7 +46,7 @@ def run_with_cpu_limit(cmd, *args):
         #   [1]: https://github.com/ClickHouse/ClickHouse/issues/32806
         args += (
             "--volume",
-            f"{online_cpu.name}:/sys/devices/system/cpu/online",
+            f"{online_cpu.name}:/sys/devices/system/cpu/online:ro",
         )
 
         return run_command_in_container(cmd, *args)
@@ -88,17 +88,20 @@ def test_jemalloc_percpu_arena():
     assert int(result) == int(1), result
 
     # should fail because of abort_conf:true
+    # This needs to pin the CPU > 0 manually because docked doesn't do it and
+    # /sys/devices/system/cpu/online is not enough.
     with pytest.raises(subprocess.CalledProcessError):
         run_with_cpu_limit(
-            'clickhouse local -q "select 1"', "--env", "MALLOC_CONF=abort_conf:true"
+            'clickhouse local -q "select 1"',
+            "--cpuset-cpus", f"{CPU_ID}",
+            "--env", "MALLOC_CONF=abort_conf:true",
         )
 
     # should not fail even with abort_conf:true, due to explicit narenas
     # NOTE: abort:false to make it compatible with debug build
     run_with_cpu_limit(
         'clickhouse local -q "select 1"',
-        "--env",
-        f"MALLOC_CONF=abort_conf:true,abort:false,narenas:{all_cpus}",
+        "--env", f"MALLOC_CONF=abort_conf:true,abort:false,narenas:{all_cpus}",
     )
 
 

